### PR TITLE
Css refactor

### PR DESCRIPTION
**Pull request by mclintprojects**
Fixes issue #25, #26, #27, #28, #29, #30, #31, #32, #33

### CODE REVIEW

File: web/src/App.vue [hunk 1]
```diff
@@ -291,34 +351,38 @@ body {
...
   to {
-    transform: translateX(-100px);
+    transform: translateX(-10rem);
     opacity: 0;
   }
 }
 
-@media screen and (max-width: 576px), (max-width: 768px) {
-  .appContainer {
+@media screen and (max-width: 57.6rem), (max-width: 76.8rem) {
```
Comment (jarikmarwede) [inline web/src/App.vue]: Why are there two queries? The first one seems to do nothing.

File: web/src/components/IdeaList.vue [hunk 2]
```diff
@@ -1,34 +1,36 @@
...
         >
           <div class="idea" @click="notifyIdeaClicked(idea, index)">
             <div>
-              <p id="idea-title" class="text-primary">
+              <p class="idea__title text--primary">
                 {{idea.title}}
                 <span v-if="isNewIdea(idea)">New</span>
               </p>
-              <p id="idea-difficulty" class="badge text-secondary">{{idea.difficulty}}</p>
+              <div>
```
Comment (jarikmarwede) [inline web/src/components/IdeaList.vue]: Why is there an empty div around the idea difficulty? It does not seem to change the appearance of it.

File: web/src/components/IdeaList.vue [hunk 3]
```diff
@@ -252,81 +256,109 @@ export default {
...
   height: 100%;
 }
-#ideas {
+
+.ideas {
   list-style-type: none;
-  margin: 0px;
-  padding: 0px;
+  margin: 0rem;
+  padding: 0rem;
```
Comment (jarikmarwede) [inline web/src/components/IdeaList.vue]: The margin and padding are already set to 0.

File: web/src/components/IdeaList.vue [hunk 4]
```diff
@@ -252,81 +256,109 @@ export default {
...
+
+.idea__title {
   font-size: var(--primaryTextSize);
-  margin: 0px 0px 4px 0px;
+  margin: 0rem 0rem 0.4rem 0rem;
   text-overflow: ellipsis;
   overflow: hidden;
   white-space: nowrap;
-  padding: 0px;
+  padding: 0rem;
```
Comment (jarikmarwede) [inline web/src/components/IdeaList.vue]: The padding is already set to 0.

File: web/src/components/IdeaList.vue [hunk 5]
```diff
@@ -252,81 +256,109 @@ export default {
...
   display: flex;
   align-items: center;
   justify-content: space-between;
 }
-#idea-title {
+
+.idea__title {
   font-size: var(--primaryTextSize);
-  margin: 0px 0px 4px 0px;
+  margin: 0rem 0rem 0.4rem 0rem;
```
Comment (jarikmarwede) [inline web/src/components/IdeaList.vue]: Can be changed to "margin-bottom: 0.4rem;"

File: web/src/components/IdeaList.vue [hunk 6]
```diff
@@ -252,81 +256,109 @@ export default {
...
   align-items: center;
 }
+
 .idea-menu-actions {
   border: none;
-  display: flex;
-  flex-flow: column;
+  display: flex !important;
+  flex-direction: column;
   padding: 0;
```
Comment (jarikmarwede) [inline web/src/components/IdeaList.vue]: This padding is overridden by popper. It overrides all of these styles because it seems to get included later and has the same specificity.

File: web/src/components/IdeaList.vue [hunk 7]
```diff
@@ -252,81 +256,109 @@ export default {
...
 }
+
+.idea-menu-actions button {
+  font-size: 1.4rem;
+  text-align: left;
+  color: rgba(0, 0, 0, 0.8);
+}
+
+.idea-menu-actions button:hover {
+  cursor: pointer;
```
Comment (jarikmarwede) [inline web/src/components/IdeaList.vue]: I would suggest just changing the default cursor for buttons to pointer.

File: web/src/components/Navbar.vue [hunk 8]
```diff
@@ -71,53 +74,69 @@ export default {
...
     flex-flow: row;
     align-items: center;
+    margin-right: 1.6rem;
   }
+
   nav > a {
-    padding: 0 1.5rem;
+    margin: 0;
+    margin-left: 2.4rem;
+    padding: 0;
```
Comment (jarikmarwede) [inline web/src/components/Navbar.vue]: The padding is already set to 0.

File: web/src/views/CategoryList.vue [hunk 9]
```diff
@@ -61,48 +75,47 @@ export default {
 };
 </script>
 
-<style scoped>
-.categoryIcon {
+<style>
```
Comment (jarikmarwede) [inline web/src/views/CategoryList.vue]: Why did you remove the scope?

File: web/src/views/CategoryList.vue [hunk 10]
```diff
@@ -61,48 +75,47 @@ export default {
...
 
-#categoryList {
-	width: 100%;
+.categories {
+  width: 100%;
   list-style-type: none;
-	margin: 0px;
-  padding: 0px;
+  margin: 0rem;
+  padding: 0rem;
```
Comment (jarikmarwede) [inline web/src/views/CategoryList.vue]: The margin and padding are already set to 0.

File: web/src/views/CategoryList.vue [hunk 11]
```diff
@@ -61,48 +75,47 @@ export default {
...
   font-size: var(--primaryTextSize);
-  margin: 0px;
 }
 
-.categoryItem {
+.category {
   display: flex;
   flex-direction: row;
-  padding: 8px 16px 8px 16px;
+  padding: 0.8rem 1.6rem 0.8rem 1.6rem;
```
Comment (jarikmarwede) [inline web/src/views/CategoryList.vue]: This can be shortened to "padding: 0.8rem 1.6rem;".

File: web/src/App.vue [hunk 12]
```diff
@@ -114,130 +114,162 @@ export default {
...
   position: absolute;
-  left: calc(50% - 36px);
-  top: calc(50% - 36px);
+  left: calc(50% - 3.6rem);
+  top: calc(50% - 3.6rem);
+  font-size: 1.8rem;
 }
 
-.appBtn {
+.button {
```
[the remark this answers is not in the document]
Comment (mclintprojects) [reply web/src/App.vue]: This is because of BEM. Also, we don't apply it to only buttons. We applied it to the Bookmarks FAB that's actually a link.

File: web/src/components/Navbar.vue (hunk 8 above)
Comment (mclintprojects) [reply web/src/components/Navbar.vue]: I am resetting the horizontal styling for the large screen view of the Navbar.

File: web/src/App.vue (hunk 12 above)
[the remark this answers is not in the document]
Comment (jarikmarwede) [reply web/src/App.vue]: Ah, ok
